src/pyFAI/gui/diffmap_widget.py

```diff
@@ -568,6 +575,35 @@ def update_processing(self, idx_file, idx_img):
             qt.QCoreApplication.processEvents()
             time.sleep(0.1)
 
+    def finish_processing(self, start_pilx=False):
+        """ close the process bar widget and the images
+
+        :param start_pilx: (str) open the pilx visualization tool with the given file
+        """
+        with self.update_sem:
+            if self.fig:
+                pyplot.close(self.fig)
+                self.fig = None
+                self.plot = None
+                self.img = None
+                self.axplt = None
+                self.aximg = None
+
+        if start_pilx and isinstance(start_pilx, str) and os.path.exists(start_pilx):
+            self.processingFinished.emit(start_pilx)
```

Comment:
No, because the visualization has to be triggered from the main thread and the processing occurs in a separate thread ... Several options have been considered with Thomas, this is one of them.

Reply:
Got it :ok_hand: